model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaRootEntity.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa;
+
+/**
+ * Interface for all root entities in the JPA storage.
+ */
+public interface JpaRootEntity {
+
+    /**
+     * Version of the JPA entity used for optimistic locking
+     */
+    int getVersion();
```

Comment:
nitpick: could there be `@Override` annotation to implementations?

Reply:
+1!